Add Additional Synthesizer UI Controls under groups

spessasus opened the pull request:
This PR adds a lot of new controller meters that were previously supported, but not added to avoid a big mess in the menu. Now users can select the category of the controllers via a dropdown menu.

Resolves #117 

Resolves #129

@TS53, what are your thoughts about this? You've been very helpful in resolving #120, so maybe you can share your opinions or suggestions? If you have the time to check, of course :-)

TS53 commented:
When I hover over "Effects", I get the description for the "Effects". When I now select one like "Volume Envelope" and hover over this one, I would expect to get a description for the "Volume Envelope" but it is the same like for "Effects". Also, after selecting "Volume Envelope", the text is cut off on my screen to "Volume Env...". If the elements get an own hover text, I would suggest to put the element name in front, so it can be read again as whole word like "Volume Envelope: here the description".

For Portamento: there is a good hover text, which says to set it to 0 to disable it. The problem is that current value is not visible and hidden by the text "Portamento ti..." on my screen size. The same for the other Portamento setting. Both hover texts should start with the exact name to make it visible on smaller screen sizes.
I would probably add the element name in front for all elements. They probably all depend on the screen size, even "Pan" etc.

After playing a bit with the settings, maybe it is good to have a reset button, to be able to quickly reset all.

Otherwise, it looks good to me!

spessasus commented:
Very helpful as always, TS :-)

Here are the changes I made:
- I've moved the effects group to the main controller, to free up some space in the channel controllers (and I think it's more logical anyway)
- I've shortened the slider titles so it shouldn't get cut off.
- I've added the full slider title to all hovers like you've suggested.
- And there's the "System Reset" at the top which does exactly what it says :-)

Now, regarding #129:
- I've made some style changes to the locked controllers: square borders, bold and italic
- I've also made it lock *automatically* when you change it. I think it would be a good way to teach users of the locking. And if they move the controller, they probably expect it to stay. You can still of course unlock it after by right-clicking again.

TS53 commented:
I think, it is very good now! Just the following things:
- It seems the locked button does work for most buttons but not for "Transpose".
- I would suggest renaming "Volume Env." to "Vol. Envelope" or "Envelope", since "Vol." is always short for Volume but "Env." is in general more often short for "Environment", I think. Wikipedia seems to have it under [Envelope](https://en.wikipedia.org/wiki/Envelope_(music)).
- "System reset" was what I wanted as reset button. But I didn't want to press it before, because I thought it resets the system (like reloading the website with soundfont again). I suggest to rename it to "Reset settings", or "Reset" or similiar so there is no confusion with the term "system".

spessasus commented:
Thanks for taking a look at the improved version! Here's what I think:

- the transpose button indeed cannot be locked. Even when you hover over it, it doesn't say "right-click to lock." This is because there's no point in locking it. The whole point of locking controllers is to prevent them from changing with a MIDI message (playing a MIDI file or using an external MIDI controller)
- That's a good idea, I will rename it. I don't think I can add the text wrapping as this would probably mess up the layout. I'll see what I can do though.
- System reset is called like that because there's a MIDI message named exactly like that and it involves that message. Though now that I'm thinking, maybe *MIDI Reset* would be a better name? Let me know!

TS53 commented:
Now the locking makes sense to me! Regarding the "System reset": I think "MIDI reset" is already better since the word targets a smaller component. For the average user without deep MIDI background I would say "Reset changes", "Reset settings" or  "Reset synthesizer settings" is most intuitive. While from a technical perspective and for people with a good overview of the MIDI format probably the exact terms are intuitively clear. I'm not sure. Maybe "Reset controllers" could also fit.

spessasus commented:
Alright, I've renamed the volume envelope and system reset to "Reset Controllers."
